components/batch-system/src/batch.rs

```diff
@@ -232,12 +226,19 @@ impl<N: Fsm, C: Fsm> Batch<N, C> {
             // failed to reschedule
             f.policy.take();
             self.normals[index] = res;
-        } else if !inplace {
+        }
+    }
+
+    /// Reclaims the slot storage if there is no FSM located at `index`. It will
+    /// alter the positions of other FSMs with index larger than `index`.
+    #[inline]
+    pub fn unsafe_reclaim(&mut self, index: usize) {
```

Comment:
It is unsafe in the sense that it can change other FSM's location index, and the index stability is a primary assumption of this container. To mark it unsafe urges caller taking responsibility to protect the invariants.

Even though it's not "Rust unsafe", but we used this word in other cases too (`unsafe_destroy_range` or `unsafe_recover`).

Reply:
unsafe_destroy_range breaks consistency and unsafe_recover break durability, so they are prefixed with `unsafe_`. For this function, we can name it `swap_reclaim` to indicate it will change order.